File: swig/python/tests/starting.py
import unittest

import os 
cwd = os.path.abspath(os.path.dirname(__file__)) 

import sys
sys.path.insert(0, cwd + "/../../../build/swig/python")

class TestSequenceFunctions(unittest.TestCase):
    
  def teststarting(self):
    import zypp
    Z = zypp.ZYppFactory.instance().getZYpp()
    assert Z
    Z.initializeTarget( zypp.Pathname("/") )
    Z.target().load();
                
if __name__ == '__main__':
  unittest.main()

File: swig/ruby/tests/starting.rb
#
# Test starting of zypp
#

$:.unshift "../../../build/swig/ruby"


# test loading of extension
require 'test/unit'

class LoadTest < Test::Unit::TestCase
  def test_loading
    require 'zypp'
    zypp = Zypp::ZYppFactory::instance.getZYpp
    assert zypp
    zconfig = Zypp::ZConfig::instance
    assert zconfig
    puts zconfig.systemArchitecture
    zconfig.setSystemArchitecture(Zypp::Arch.new("i686"))
    puts zconfig.systemArchitecture
  end
end
